File: cpp/7kyu/population.cpp
class Arge
{
  public:
  static int nbYear(int p0, double percent, int aug, int p)
    {
    int year(0);
    double more(p0);
    double pourcent = percent / 100;
    while (more < p)
    {
      more = (more + more * pourcent + aug);
      year +=1;
    }
    return year;

    }

};

File: population.php
function nbYear($p0, $percent, $aug, $p)
{
  $year = 0;
  $more = $p0;
  $percent /= 100;
  do
  {
   $more = ((int)($more + $more * $percent + $aug));
   $year++;
  } while ($more < $p);
  return $year;
}